### PR TITLE
scheduler: fix scheduler save config

### LINKED ISSUE #6897: evict-leader-scheduler lost config after PD leader transfer
### What did you do?

```console
$ tiup playground v7.1.1 --mode tikv-slim --pd 3 --kv 3 --without-monitor
```

```console
$ # first check what stores are available
$ tiup ctl:v7.1.1 pd store --jq .stores[].store.id
1
3
2

$ # add the evict-leader-scheduler
$ tiup ctl:v7.1.1 pd scheduler add evict-leader-scheduler 1
$ tiup ctl:v7.1.1 pd scheduler add evict-leader-scheduler 2
$ tiup ctl:v7.1.1 pd scheduler add evict-leader-scheduler 3
$ tiup ctl:v7.1.1 pd scheduler config evict-leader-scheduler
{"store-id-ranges":{"1":[{"start-key":"","end-key":""}],"2":[{"start-key":"","end-key":""}],"3":[{"start-key":"","end-key":""}]}}

$ # transfer leader 
$ tiup ctl:v7.1.1 pd member leader show | jq .name
"pd-2"
$ tiup ctl:v7.1.1 pd member leader transfer pd-0

$ # wait until the schedulers are ready...
$ tiup ctl:v7.1.1 pd scheduler config evict-leader-scheduler
[404] scheduler not found
...
$ tiup ctl:v7.1.1 pd scheduler config evict-leader-scheduler
{"store-id-ranges":{"1":[{"start-key":"","end-key":""}],"2":[{"start-key":"","end-key":""}],"3":[{"start-key":"","end-key":""}]}}

$ # transfer leader again
$ tiup ctl:v7.1.1 pd member leader transfer pd-1
$ tiup ctl:v7.1.1 pd scheduler config evict-leader-scheduler
[404] scheduler not found
...
$ tiup ctl:v7.1.1 pd scheduler config evict-leader-scheduler # ?!
{"store-id-ranges":{"1":[{"start-key":"","end-key":""}]}}
```

### What did you expect to see?

The `evict-leader-scheduler` retains the config showing all stores 1, 2, 3.

### What did you see instead?

After the second transfer the scheduler can only see the first added store 1.

### What version of PD are you using (`pd-server -V`)?

v6.5.3, v7.1.1

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #6897

### What is changed and how does it work?

```commit-message
```
Nowadays we have 2 configs setting way
We need to make sure that create just means create and not further save the configuration

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm

File: pkg/mcs/scheduling/server/cluster.go [hunk 1]
```diff
@@ -252,7 +252,7 @@ func (c *Cluster) updateScheduler() {
 					zap.Strings("scheduler-args", scheduler.Args))
 				continue
 			}
-			if err := schedulersController.AddScheduler(s, scheduler.Args...); err != nil {
+			if err := schedulersController.AddScheduler(c.storage, s, scheduler.Args...); err != nil {
```
Comment (JmPotato) [inline pkg/mcs/scheduling/server/cluster.go:255]: The controller has a storage field internal, we don't need to pass it all around.

Comment (HuSharp) [reply pkg/mcs/scheduling/server/cluster.go]: Thx for ur information!

**Review by JmPotato — APPROVED** [review]

**Pull request merged**